rosbag2_cpp/include/rosbag2_cpp/writer.hpp

```diff
@@ -144,14 +144,38 @@ class ROSBAG2_CPP_PUBLIC Writer final
    * \param topic_name the string of the topic this messages belongs to
    * \param type_name the string of the type associated with this message
    * \param time The time stamp of the message
-   * \throws runtime_error if the Writer is not open.
+   * \throws runtime_error if the Writer is not open or duplicating message is failed.
    */
+  [[deprecated(
+    "Use write(std::shared_ptr<rclcpp::SerializedMessage> message," \
+    " const std::string & topic_name," \
+    " const std::string & type_name," \
+    " const rclcpp::Time & time) instead."
+  )]]
```

Comment:
Line 199 in this file is causing the Windows warning - since the "non-serialized ROS message" template function calls this "reference version".

I think that the solution is to create a shared_ptr to the SerializedBagMessage, instead of a local instance.

Something like:

```
  template<class MessageT>
  void write(
    const MessageT & message,
    const std::string & topic_name,
    const rclcpp::Time & time)
  {
    auto serialized_msg = std::make_shared<rclcpp::SerializedMessage>();
    rclcpp::Serialization<MessageT> serialization;
    serialization.serialize_message(&message, serialized_msg.get());
    return write(serialized_msg, topic_name, rosidl_generator_traits::name<MessageT>(), time);
```